clang/lib/Sema/SemaOpenMP.cpp

```diff
@@ -16373,21 +16359,20 @@ OMPClause *SemaOpenMP::ActOnOpenMPOrderClause(
         << getOpenMPClauseName(OMPC_order);
     return nullptr;
   }
-  if (getLangOpts().OpenMP >= 51) {
```

Comment:
Original implementation did not call `setRegionHasOrderConcurrent(true)` for OpenMP 5.0, causing Clang to ignore all order-concurrent nestable rules for `-fopenmp-version=50`.